### DISCUSSION TITLE
garden/how-to-install-appwrite-on-synology-nas

### DISCUSSION THREAD

**Discussion by giscus** [kirawat/kirawat.me#11] [Comments]
# garden/how-to-install-appwrite-on-synology-nas

https://kirawat.me/garden/how-to-install-appwrite-on-synology-nas

**Comment by HungryBytesApps** [comment]
Was looking for a way to install Appwrite on a Synology NAS, thank you very much for this tutorial! One question: how do you perform any updates on Appwrite? Normally I get notificated in Container Manager when there is an update available and get the option to update an image.

**Reply by kirawat** [reply]
### Here is how I managed to update Appwrite to the new version (from 1.4.13 to 1.5.7)

1. In Container Manager → Project, stop currently running `appwrite` project by click `Action` → `Stop`.

2. In File Manager, make a backup of your `/volume1/docker/appwrite/storage`. Copy and paste it somewhere safe, in case something goes wrong during update.

3. Prevent `appwrite-mariadb` from restart itself by open the current `docker-compose.yml` file and edit `mariadb:` as follow then save the file:

    ```yaml
    mariadb:
      image: mariadb:10.7 # fix issues when upgrading using: mysql_upgrade -u root -p
      container_name: appwrite-mariadb
      <<: *x-logging*
      restart: "no" # <- previously: unless-stopped
    ```

4. In Container Manager → Project, click `Action` → `Build`. Wait until `appwrite` project is running.

5. You need to gain access to your Synology NAS via SSH. Go to Control Panel → Terminal & SNMP → Terminal. Check `Enable SSH service` and take note on `Port` number. For this example, it will be `22`.

6. Open Terminal and use `ssh yourNASusername@yourNASip -p22` command to gain access to your Synology NAS via SSH. Enter your NAS password as requires.

7. Shutdown `appwrite-mariadb` using the following command:

    ```bash
    sudo docker exec -it appwrite-mariadb mysqladmin -u root -p shutdown
    ```
    
    It will asks for your NAS password due to the use of `sudo`. Enter the password and wait for `appwrite-mariadb` to shutdown and stopped. Verify that `appwrite-mariadb` is actually stop running using `docker ps` command.

8. Once `appwrite-mariadb` is completed shutdown and stopped, go to Container Manager on your Synology NAS → Project, then stop the `appwrite` project by click `Action` → `Stop`.

9. When the project is stopped, click `Action` → `Clean`. Don't worry, your current data are safe within `/volume1/docker/appwrite/storage/`.

10. Download latest Appwrite base [`docker-compose.yml`](https://appwrite.io/install/compose) and [`.env`](https://appwrite.io/install/env) files.

11. Open the new `docker-compose.yml` file in code editor, under `traefik:` edit the following from:

    ```yaml
    ports:
      - 80:80
      - 443:443
    ```
    
    To
    
    ```yaml
    ports:
      - 8080:80
      - 8443:443
    ```
    
    For this example, port `8080` and `8443` is use to gain access to Appwrite. You can change it to other ports that you want to use.
    
    Replace all the paths in `volumes:` of each containers:
    
    - From `appwrite-config`
      To `/volume1/docker/appwrite/storage/config`
      
    - From `appwrite-certificates`
      To `/volume1/docker/appwrite/storage/certificates`
      
    - From `appwrite-uploads`
      To `/volume1/docker/appwrite/storage/uploads`
      
    - From `appwrite-cache`
      To `/volume1/docker/appwrite/storage/cache`
      
    - From `appwrite-functions`
      To `/volume1/docker/appwrite/storage/functions`
      
    - From `appwrite-builds`
      To `/volume1/docker/appwrite/storage/builds`
      
    - From `appwrite-mariadb`
      To `/volume1/docker/appwrite/storage/mariadb`
      
    - From `appwrite-redis`
      To `/volume1/docker/appwrite/storage/redis`
      
    Remove or comment out the unused named volumes:
    
    ```yaml
    # volumes:
    #   appwrite-mariadb:
    #   appwrite-redis:
    #   appwrite-cache:
    #   appwrite-uploads:
    #   appwrite-certificates:
    #   appwrite-functions:
    #   appwrite-builds:
    #   appwrite-config:
    ```

12. Open and edit `.env` file as necessary according to your preferences.

13. Replace the old `docker-compose.yml` and `.env` files in `/volume1/docker/appwrite` with the new one.

14. In Container Manager → Project, click `Action` → `Build`. Wait until it's finish building and the project is up and running. Wait until the database updating itself to the newer version. You can check the progress in Terminal (using SSH) by using `docker logs appwrite-mariadb` command.

15. Once the database is updated. You can check if it's success by using the following command:

    ```bash
    sudo docker exec -it appwrite-mariadb bash -c "mysql_upgrade -u root -p"
    ```
    
    It should asks for your NAS password, and follow by `root` password for MariaDB, which is set according to `_APP_DB_ROOT_PASS=` in `.env` file.
    
    You should get the following message:
    
    ```
    This installation of MariaDB is already upgraded to 10.11.8-MariaDB.
    There is no need to run mysql_upgrade again for 10.11.8-MariaDB.
    You can use --force if you still want to run mysql_upgrade
    ```
    
    That mean it's ready to continue the [migration process](https://appwrite.io/docs/advanced/self-hosting/update#running-the-migration) in the next step.
    
16. Run the following command to migrate current database to the new version of Appwrite:

    ```bash
    sudo docker exec appwrite migrate
    ```
    
    The time it would takes for data migration to completed depending on the amount of data your Appwrite instance contains.
    
    Once the data migration is completed, you can now access the new version of Appwrite via `https://[YOUR-NAS-IP-ADDRESS]:8443` or other ports you set in `docker-compose.yml` file.
